clash-cores/src/Clash/Cores/Xilinx/Ethernet/Gmii.hs

```diff
@@ -0,0 +1,136 @@
+  Signal sgmii625 Lvds ->
+  -- | GMII input from the MAC.
+  Signal gmiiTx125 Gmii ->
+  -- | Output record, see 'BridgeOutput'.
+  BridgeOutput gmiiRx125 sgmii625
+gmiiSgmiiBridge refClk refRst  signalDetect bridgeConfig anConfig anRestart lvdsIn gmiiTx = BridgeOutput{..}
+ where
+  ( lvdsOutP
+    , lvdsOutN
+    , gmiiRxD
+    , gmiiRxDv
+    , gmiiRxEr
+    , bridgeClk125
+    , activeHighRxRst
+    , fmap fromStatusVector -> bridgeStatusVector
+    ) = gmiiSgmiiBridgeBb clockP clockN activeHighRefRst signalDetect
+      (toConfigVector <$> bridgeConfig) anConfig anRestart lvdsInP lvdsInN gmiiTxD
+      gmiiTxEn gmiiTxEr
+
+  activeHighRefRst = unsafeToActiveHigh refRst
+  bridgeRst125 = unsafeFromActiveHigh activeHighRxRst
+  bridgeLvdsOut = Lvds <$> lvdsOutP <*> lvdsOutN
+  (DiffClock clockP clockN) = refClk
+  (lvdsInP, lvdsInN) = (fmap lvdsP lvdsIn, fmap lvdsN lvdsIn)
+  (gmiiTxD, gmiiTxEn, gmiiTxEr) = unbundle $ fmap (\Gmii{..} -> (gmiiData, gmiiValid, gmiiError)) gmiiTx
+  bridgeGmiiRx = Gmii <$> gmiiRxD <*> gmiiRxDv <*> gmiiRxEr
+
+-- | Black box for the GMII to SGMII bridge.
+gmiiSgmiiBridgeBb ::
+  -- | P channel of the reference clock coming from the Phy.
+  Clock sgmii625 ->
+  -- | N channel of the reference clock coming from the Phy.
+  ClockN sgmii625 ->
+  -- | Asynchronous active high reset for the bridge.
+  Signal asyncDom Bool->
+  -- | Signal detect from the Phy.
+  -- Either connect to the Phy's signal detect or use a constant 'True`,
+  -- otherwise the link will never come up.
+  Signal asyncDom Bool ->
+  -- | Configuration for the bridge.
+  Signal gmiiRx125 ConfigVector ->
+  -- | Auto negotiation configuration for the bridge.
+  Signal gmiiRx125 AutoNegConfig ->
+
+  -- | Restart auto negotiation.
+  Signal gmiiRx125 Bool ->
+  -- | P channel of the LVDS input from the Phy.
+  Signal sgmii625 Bit ->
+  -- | N channel of the LVDS input from the Phy.
+  Signal sgmii625 Bit ->
+  -- | GMII data input from the MAC.
+  Signal gmiiTx125 (BitVector 8) ->
+  -- | GMII valid input from the MAC.
+  Signal gmiiTx125 Bit ->
+  -- | GMII error input from the MAC.
+  Signal gmiiTx125 Bit ->
+  -- |
+  -- 1. P channel of the LVDS output to the Phy.
+  -- 2. N channel of the LVDS output to the Phy.
+  -- 3. GMII data output to the MAC.
+  -- 4. GMII valid output to the MAC.
+  -- 5. GMII error output to the MAC.
+  -- 6. Clock output for the 125 MHz domain.
+  -- 7. Active high reset output for the 125 MHz domain.
+  -- 8. Status vector.
+  ( Signal sgmii625 Bit
+  , Signal sgmii625 Bit
+  , Signal gmiiRx125 (BitVector 8)
+  , Signal gmiiRx125 Bit
+  , Signal gmiiRx125 Bit
+  , Clock gmiiRx125
+  , Signal gmiiRx125 Bool
+  , Signal gmiiRx125 StatusVector
+  )
+
+-- Can't use `deepErrorX' because one of the results is a clock.
+gmiiSgmiiBridgeBb !_ !_  !_ !_ !_ !_ !_ !_ !_ !_ !_ !_ =
+  (undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined)
+
+{-# OPAQUE gmiiSgmiiBridgeBb #-}
```

Comment:
```suggestion
{-# CLASH_OPAQUE gmiiSgmiiBridgeBb #-}
```

(and don't forget `Language CPP`)

More in general, there are still things preventing succesful compilation, see CI.

Reply:
I get an `Unrecognised pragma` warning on both `LANGUAGE CPP` and the `CLASH_OPAQUQE` annotation. Do you perhaps know why?

Reply:
It works for me, you must have gotten a detail wrong. I tested it with GHC 9.6.4, I'm now testing with 9.0.2 to be completely certain, but I'd be flabbergasted if it failed.

```diff
--- a/clash-cores/src/Clash/Cores/Xilinx/Ethernet/Gmii.hs
+++ b/clash-cores/src/Clash/Cores/Xilinx/Ethernet/Gmii.hs
@@ -9,8 +9,10 @@
 -}
 
 {-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE RecordWildCards #-}
 {-# LANGUAGE ViewPatterns #-}
+
 {-# OPTIONS_GHC -fconstraint-solver-iterations=10 #-}
 
 module Clash.Cores.Xilinx.Ethernet.Gmii where
@@ -131,6 +133,6 @@ gmiiSgmiiBridgeBb ::
 gmiiSgmiiBridgeBb !_ !_  !_ !_ !_ !_ !_ !_ !_ !_ !_ !_ =
   (undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined)
 
-{-# OPAQUE gmiiSgmiiBridgeBb #-}
+{-# CLASH_OPAQUE gmiiSgmiiBridgeBb #-}
 {-# ANN gmiiSgmiiBridgeBb hasBlackBox #-}
 {-# ANN gmiiSgmiiBridgeBb (veriBinaryPrim) #-}
```

Reply:
I get
```hs
src/Clash/Cores/Xilinx/Ethernet/Gmii.hs:12:1: warning: [-Wunrecognised-pragmas]
    Unrecognised pragma
   |
12 | {-# LANGAUGE CPP #-}
   | ^^^
src/Clash/Cores/Xilinx/Ethernet/Gmii.hs:215:1: warning: [-Wunrecognised-pragmas]
    Unrecognised pragma
    |
215 | {-# CLASH_OPAQUE gmiiSgmiiBridgePrim #-}
    | ^^^
```
Let's debug this face to face :)

Reply:
My face is 170 km away, but that won't be necessary. langAUge isn't language. It's perhaps an indicator of the speed of your local area network, but it's not a word nor is it the correct word.

:-P